src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3626,16 +3631,256 @@ bool isInstanceConstructor(out MethodSymbol constructorSymbol)
             }
         }
 
-        internal virtual BoundExpressionStatement BindConstructorInitializer(ConstructorInitializerSyntax initializer, BindingDiagnosticBag diagnostics)
+#nullable enable
+        [return: NotNullIfNotNull("initializer")]
+        internal virtual BoundExpressionStatement? BindConstructorInitializer(ConstructorInitializerSyntax? initializer, SyntaxNode? ctorSyntax, BindingDiagnosticBag diagnostics)
         {
-            BoundExpression initializerInvocation = GetBinder(initializer).BindConstructorInitializer(initializer.ArgumentList, (MethodSymbol)this.ContainingMember(), diagnostics);
+            Debug.Assert(initializer != null || ctorSyntax != null);
+            BoundExpression? initializerInvocation = initializer is null
+                ? BindImplicitConstructorInitializer((MethodSymbol)this.ContainingMember(), diagnostics, Compilation)
+                : GetRequiredBinder(initializer).BindConstructorInitializer(initializer.ArgumentList, (MethodSymbol)this.ContainingMember(), diagnostics);
+            Debug.Assert(initializerInvocation is not null || initializer is null);
+
+            if (initializerInvocation is null)
+            {
+                return null;
+            }
+
             //  Base WasCompilerGenerated state off of whether constructor is implicitly declared, this will ensure proper instrumentation.
             Debug.Assert(!this.ContainingMember().IsImplicitlyDeclared);
-            var constructorInitializer = new BoundExpressionStatement(initializer, initializerInvocation);
+            var constructorInitializer = new BoundExpressionStatement(initializer ?? ctorSyntax!, initializerInvocation) { WasCompilerGenerated = ((MethodSymbol)ContainingMember()).IsImplicitlyDeclared };
             Debug.Assert(initializerInvocation.HasAnyErrors || constructorInitializer.IsConstructorInitializer(), "Please keep this bound node in sync with BoundNodeExtensions.IsConstructorInitializer.");
             return constructorInitializer;
         }
 
+        /// <summary>
+        /// Bind the implicit constructor initializer of a constructor symbol.
+        /// </summary>
+        /// <param name="constructor">Constructor method.</param>
+        /// <param name="diagnostics">Accumulates errors (e.g. access "this" in constructor initializer).</param>
+        /// <param name="compilation">Used to retrieve binder.</param>
+        /// <returns>A bound expression for the constructor initializer call.</returns>
+        internal static BoundExpression? BindImplicitConstructorInitializer(
+            MethodSymbol constructor, BindingDiagnosticBag diagnostics, CSharpCompilation compilation)
+        {
+            if (constructor.MethodKind != MethodKind.Constructor || constructor.IsExtern)
+            {
+                return null;
+            }
```

Comment:
The only non-nullability change in these methods. The intent is that static constructors and extern constructors do not get implicit initializers.

Reply:
I'll add a test demonstrating behavior of this, but it works as is.